opensearch/chart/templates/tests/test-opensearch-config.yaml

```
@@ -0,0 +1,123 @@
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-test
+  namespace: {{ .Release.Namespace }}
+  labels:
+    type: integration-test
+    {{- include "opensearch-cluster-local.labels" . | nindent 4 }}
+data:
+  run.sh: |-
+
+    #!/usr/bin/env bats
+
+    load "/usr/lib/bats/bats-detik/utils"
+    load "/usr/lib/bats/bats-detik/detik"
+
+    DETIK_CLIENT_NAME="kubectl"
+
+    @test "Verify successful deployment and running status of the operator pod" {
+        try "at most 12 times every 10s to get pods named 'opensearch-operator-controller-manager-[a-z0-9]+-[a-z0-9]+' and verify that '.status.phase' is 'Running'"
+    }
+
+    @test "Verify successful deployment of the operator service" {
+        verify "there is 1 service named '^opensearch-operator-controller-manager-metrics-service$'"
+    }
+
+    @test "Verify presence of OpenSearchCluster resource" {
+      [ "$(kubectl get opensearchclusters.opensearch.opster.io -n {{ .Release.Namespace }} --no-headers | wc -l)" -ge 1 ]
+    }
+
+    @test "Verify presence of OpenSearchIndexTemplate resource" {
+      [ "$(kubectl get opensearchindextemplates.opensearch.opster.io -n {{ .Release.Namespace }} --no-headers | wc -l)" -ge 1 ]
+    }
+
+    @test "Verify presence of OpenSearchRole resource" {
+      [ "$(kubectl get opensearchroles.opensearch.opster.io -n {{ .Release.Namespace }} --no-headers | wc -l)" -ge 1 ]
+    }
+
+    @test "Verify presence of OpenSearchUser resource" {
+      [ "$(kubectl get opensearchusers.opensearch.opster.io -n {{ .Release.Namespace }} --no-headers | wc -l)" -ge 1 ]
+    }
+
+    @test "Verify presence of OpenSearchUserRoleBinding resource" {
+      [ "$(kubectl get opensearchuserrolebindings.opensearch.opster.io -n {{ .Release.Namespace }} --no-headers | wc -l)" -ge 1 ]
+    }
+
+    @test "Verify presence of OpenSearchISMPolicy resource" {
+      [ "$(kubectl get opensearchismpolicies.opensearch.opster.io -n {{ .Release.Namespace }} --no-headers | wc -l)" -ge 1 ]
+    }
+
+    @test "Verify presence of OpenSearchComponentTemplate resource" {
+      [ "$(kubectl get opensearchcomponenttemplates.opensearch.opster.io -n {{ .Release.Namespace }} --no-headers | wc -l)" -ge 1 ]
+    }
+
+    @test "Verify presence of OpenSearchTenant resource" {
+      [ "$(kubectl get opensearchtenants.opensearch.opster.io -n {{ .Release.Namespace }} --no-headers | wc -l)" -ge 0 ]
+    }
+
+    @test "Verify OpenSearch client service exists" {
+        verify "there is 1 service named '{{ .Release.Name }}-client-external'"
+    }
+
+    @test "Verify OpenSearch main, data, and bootstrap pods are running" {
+      tries=12
+      interval=10
+      for i in $(seq 1 $tries); do
+        all_running=true
+        kubectl get pods -n {{ .Release.Namespace }} | grep -E 'opensearch-(main|data|bootstrap|client)-' | awk '{print $1,$3}' | while read name status; do
+          if [[ "$status" != "Running" ]]; then
+            all_running=false
+            break
+          fi
```

You are never using the `name` variable. Maybe add it for some flavour?
```suggestion
        kubectl get pods -n {{ .Release.Namespace }} | grep -E 'opensearch-(main|data|bootstrap|client)-' | awk '{print $1,$3}' | while read name status; do
          if [[ "$status" != "Running" ]]; then
            all_running=false
            echo "$name is not running"
            break
          fi
```